internal/components/single_endpoint.go

```diff
@@ -55,64 +48,38 @@ func (g *SingleEndpointConfig) GetPortNum() (int32, error) {
 	return UnsetPort, PortNotFoundErr
 }
 
-// SingleEndpointParser is a special parser for a generic receiver that has an endpoint or listen_address in its
-// configuration. It doesn't self-register and should be created/used directly.
-type SingleEndpointParser struct {
-	name string
-
-	svcPort *corev1.ServicePort
+func SilentSingleEndpointParser(logger logr.Logger, name string, o *Option, singleEndpointConfig *SingleEndpointConfig) ([]corev1.ServicePort, error) {
```

Comment:
nit: I don't really like the names of these functions. For example, this one doesn't return a parser struct, it returns the actual port definitions. So it'd make sense for it to have a more typical verb-based name, like `ParseSingleEndpointSilent`. Same for the other names in this file.